Support variables with dot in python templater (#5872)
sqlfluff/sqlfluff: docs/source/configuration.rst

    SELECT * FROM my_table

Complex Python Variable Templating
""""""""""""""""""""""""""""""""""""

`Python string formatting`_ supports accessing object attributes
via dot notation (e.g. :code:`{foo.bar}`).  However, since we cannot create Python
objects within configuration files, we need a workaround in order to provide

> dummy values to render templates containing these values.  The SQLFluff
> python templater will temporarily convert any template variable
> containing a "." with a dictionary lookup using the fixed context key :code:`sqlfluff`:

Reviewer: I think the use of `temporarily` here is misleading. Perhaps this might be clearer:

```suggestion
dummy values to render templates containing these values.  The SQLFluff
python templater will interpret any variable containing a "." as a
dictionary lookup on the *magic* fixed context key :code:`sqlfluff`.
```
